### PR TITLE
Edit session

### PR DESCRIPTION
**Pull request by michalrazny2**
Task: https://projects.invisionapp.com/d/main?origin=v7#/console/19745972/414476833/preview?scrollOffset=4154
Rendering: https://projects.invisionapp.com/d/main?origin=v7#/console/19745972/414476833/preview?scrollOffset=4208

For sure will need some changes and clean up, but I share so you are able to see my progress :)

TODOs for now on this branch:

- [x] the request should be sent to the correct endpoint
- [x] make sure it works fine for both mobile and fixed sessions
- [x] ensure what is happening if the user edits session when the network is gone - we need to do it in sync service (sessions updated locally should be updated on server)
- [x] some null assertions to delete and misplaced functions to move
- [x] a couple minor problems (like '[]' showing at tags_input after ArrayList.toString() usage)
- [x] form fields - should be changeable with next
- [x] test
- [x] refactor

### CODE REVIEW

File: app/src/main/java/io/lunarlogic/aircasting/models/Session.kt [hunk 1]
```diff
@@ -253,4 +257,9 @@ class Session(
         formatter.timeZone = TimeZone.getDefault()
         return formatter
     }
+
+    fun sessionEdited(newName: String, newTags: ArrayList<String>){
```
Comment (mturnau) [inline app/src/main/java/io/lunarlogic/aircasting/models/Session.kt:261]: let's rename the method to something more specifig, e.g. setNameAndTags or or setSessionDetails editSessionDetails

File: app/src/main/java/io/lunarlogic/aircasting/sensor/SessionManager.kt [hunk 2]
```diff
@@ -122,6 +127,13 @@ class SessionManager(private val mContext: Context, private val apiService: ApiS
         }
     }
 
+    private fun editSession(event: EditSessionEvent) {
+        DatabaseProvider.runQuery {
+                sessionsRespository.update(event.session)
```
Comment (mturnau) [inline app/src/main/java/io/lunarlogic/aircasting/sensor/SessionManager.kt:132]: I think we don't have endpoint implemented for updating session.

- Check the old app, SessionDriver l 125
- ensure editing fixed and mobile sessions look the same
- new method in ApiService

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionsController.kt [hunk 3]
```diff
@@ -104,4 +112,33 @@ abstract class SessionsController(
             mDownloadMeasurementsService.downloadMeasurements(session, finallyCallback)
         }
     }
+
+    override fun onEditDataPressed() { // handling buttons in EditSessionBottomSheet
+        val editData = dialog?.editDataConfirmed()
+        if(editData == null){
```
Comment (mturnau) [inline app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionsController.kt:118]: add some spaces ;)

**Comment by mturnau** [comment]
I still haven't figured out ow to fix the test (scrolling not working on the bottom sheet? But I'll merge it for now

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/EditSessionBottomSheet.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        val closeButton = view?.findViewById<ImageView>(R.id.close_button)
+        closeButton?.setOnClickListener {
+            mListener.onCancelPressed()
+        }
+
+        return view
+    }
+
+    fun editDataConfirmed(): Session{
+        val sessionName = view?.session_name_input?.text.toString().trim()
```
[the remark this answers is not in the document]
Comment (lesniakania) [reply app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/EditSessionBottomSheet.kt:54]: Yeah, I think part of this method should be in a controller. I'm looking at this PR and I think I will just make some small tweaks and share another PR with you :)

**Pull request merged**